io/net/virt-net/network_virtualization.py

```diff
@@ -271,22 +310,31 @@ def test_clientfailover(self):
         except CmdError as details:
             self.log.debug(str(details))
             self.fail("Client initiated Failover for Network virtualized \
-                       device has failed")
+                      device has failed")
 
     def test_backingdevremove(self):
         '''
         Removing Backing device for Network virtualized device
         '''
+        if self.check_slot_availability():
+            self.fail("Slot does not exist")
+        self.update_backing_devices()
         pre_remove = self.backing_dev_count()
-        self.backing_dev_add_remove('remove')
+        for count in range(1, self.backingdev_count):
+            self.backing_dev_add_remove('remove', count)
         post_remove = self.backing_dev_count()
-        if pre_remove - post_remove != 1:
+        if pre_remove - post_remove + 1 != self.backingdev_count:
```

Comment:
can this be move in one variable as it increase read ability of code , just a suggestion

Reply:
any suggestions @PraveenPenguin ?

Reply:
`device_count = pre_remove - post_remove` can this fine

Reply:
Ok, will do, since you feel the need for it :-)